test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ContactDetailsConfirmAddressControllerSpec.scala

```diff
@@ -0,0 +1,196 @@
+import uk.gov.hmrc.plasticpackagingtax.registration.views.html.confirm_address
+import uk.gov.hmrc.play.bootstrap.tools.Stubs.stubMessagesControllerComponents
+
+class ContactDetailsConfirmAddressControllerSpec extends ControllerSpec {
+  private val page = mock[confirm_address]
+  private val mcc  = stubMessagesControllerComponents()
+
+  private val controller =
+    new ContactDetailsConfirmAddressController(authenticate = mockAuthAction,
+                                               mockJourneyAction,
+                                               mockRegistrationConnector,
+                                               mcc = mcc,
+                                               page = page
+    )
+
+  private val anAddress =
+    Address(addressLine1 = "Address Line 1", townOrCity = "townOrCity", postCode = "LS3 3UJ")
+
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+    when(page.apply(any[Form[ConfirmAddress]])(any(), any())).thenReturn(HtmlFormat.empty)
+  }
+
+  override protected def afterEach(): Unit = {
+    reset(page)
+    super.afterEach()
+  }
+
+  "Primary Contact Details Confirm Address Controller" should {
+
+    "return 200" when {
+
+      "user is authorised and display page method is invoked" in {
+        authorizedUser()
+        val result = controller.displayPage()(getRequest())
+
+        status(result) mustBe OK
+      }
+
+      "user is authorised, a registration already exists and display page method is invoked" in {
+        authorizedUser()
+        mockRegistrationFind(
+          aRegistration(
+            withPrimaryContactDetails(
+              PrimaryContactDetails(useRegisteredAddress = true, address = Some(anAddress))
+            )
+          )
+        )
+        val result = controller.displayPage()(getRequest())
+
+        status(result) mustBe OK
+      }
+    }
+
+    forAll(Seq(saveAndContinueFormAction, saveAndComeBackLaterFormAction)) { formAction =>
+      "return 303 (OK) for " + formAction._1 when {
+        val registrationWithoutAddress = aRegistration(
+          withPrimaryContactDetails(
+            PrimaryContactDetails(
+              fullName = Some(FullName(firstName = "Jack", lastName = "Gatsby")),
+              jobTitle = Some("Developer"),
+              email = Some("test@test.com"),
+              phoneNumber = Some("0203 4567 890"),
+              address = None
+            )
+          )
+        )
+        "user accepts the registered address" in {
+          authorizedUser()
+          mockRegistrationFind(registrationWithoutAddress)
+          mockRegistrationUpdate(registrationWithoutAddress)
+
+          val correctForm = Seq("useRegisteredAddress" -> "yes", formAction)
+          val result      = controller.submit()(postJsonRequestEncoded(correctForm: _*))
+
+          status(result) mustBe SEE_OTHER
+
+          //TODO
+//          modifiedRegistration.primaryContactDetails.address mustBe Some(anAddress)
```

Comment:
probably can be removed

Reply:
yes this was just a reminder of what I needed to test.
This has been addressed

Reply:
This was to remind of something I needed to test.